mam/v2/alloc.c

```diff
@@ -0,0 +1,33 @@
+
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ [ITSec Lab]
+
+ *
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+/*!
+\file alloc.c
+\brief MAM2 memory allocator.
+*/
+
+#include "mam/v2/alloc.h"
+
+#include <stdlib.h>
+
+void *mam2_alloc(ialloc *a, size_t s) { return malloc(s); }
+
+void mam2_free(ialloc *a, void *p) {
+  if (p) free(p);
```

Comment:
Quoting the C standard, 7.20.3.2/2 from ISO-IEC 9899:
> void free(void *ptr);
> If ptr is a null pointer, no action occurs.

Don't check for NULL, it only adds more dummy code to read and is thus a bad practice.